src/hooks/menuItemsHooks/useProblemPaneMenuItems.tsx

```diff
@@ -0,0 +1,107 @@
+import {useMemo} from 'react';
+import MonacoEditor, {monaco} from 'react-monaco-editor';
+
+import {TabsProps} from 'antd';
+
+import {useAppSelector} from '@redux/hooks';
+import {activeResourceStorageSelector} from '@redux/selectors/resourceMapSelectors';
+import {
+  problemFilePathAndRangeSelector,
+  problemResourceIdAndRangeSelector,
+  useValidationSelector,
+} from '@redux/validation/validation.selectors';
+
+import {Monaco} from '@components/molecules';
+
+import {KUBESHOP_MONACO_THEME} from '@utils/monaco';
+
+import {getRuleForResult} from '@monokle/validation';
+import {ResourceSelection} from '@shared/models/selection';
+
+const monacoOptions: monaco.editor.IStandaloneEditorConstructionOptions = {
+  readOnly: true,
+  minimap: {
+    enabled: false,
+  },
+  lineNumbers: 'off',
+};
+
+export function useProblemPaneMenuItems(width: number, height: number) {
+  const activeStorage = useAppSelector(activeResourceStorageSelector);
+  const lastResponse = useValidationSelector(state => state.lastResponse);
+  const selectedProblem = useValidationSelector(state => state.validationOverview.selectedProblem?.problem ?? null);
+  const selectedProblemMonacoData = useValidationSelector(state => {
```

Comment:
this selector always returns a new object when the state changes so it will trigger a re-render, you could send `fastDeepEqual` as the second parameter to `useAppSelector` to check if the object has changed or not